tests/storage/test_get_cookies.py

```diff
@@ -0,0 +1,101 @@
+#  Copyright 2023 Google LLC.
+#  Copyright (c) Microsoft Corporation.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import pytest
+from test_helpers import (AnyExtending, eval_expression, execute_command,
+                          goto_url)
+
+
+@pytest.mark.asyncio
+async def test_get_cookies(websocket, context_id, example_url):
+    await goto_url(websocket, context_id, example_url)
+    await eval_expression(websocket, context_id, 'document.cookie = "foo=bar"')
+    resp = await execute_command(websocket, {
+        'method': 'proto.storage.getCookies',
+        'params': {
+            'cookie': {}
+        }
+    })
+
+    assert resp == {
+        'cookies': [{
+            'domain': example_url.split('://')[1].split('/')[0].split(':')[0],
+            'expiry': -1,
+            'httpOnly': False,
+            'name': 'foo',
+            'path': '/',
+            'sameSite': 'none',
+            'secure': False,
+            'size': 6,
+            'value': {
+                'type': 'string',
+                'value': 'bar',
+            },
+        }, ],
+        'partitionKey': {},
+    }
+
+
+async def assert_cookie_filter(websocket, context_id, filter,
+                               expected_cookie_name, expected_cookie_value,
+                               expected_cookie_domain):
+    resp = await execute_command(websocket, {
+        'method': 'proto.storage.getCookies',
+        'params': {
+            'cookie': filter
+        }
+    })
+    assert resp == {
+        'cookies': [
+            AnyExtending({
+                'domain': expected_cookie_domain,
+                'name': expected_cookie_name,
+                'value': {
+                    'type': 'string',
+                    'value': expected_cookie_value,
+                },
+            }),
+        ],
+        'partitionKey': {},
+    }
+
+
+@pytest.mark.asyncio
+async def test_get_cookies_filter(websocket, context_id, another_context_id,
+                                  example_url, another_example_url):
+    example_domain = example_url.split('://')[1].split('/')[0].split(':')[0]
```

Comment:
is there a built-in URL parse in Python? https://docs.python.org/3/library/urllib.parse.html ?

Reply:
at least a helper for url to domain parsing would seem useful

Reply:
Good point. Done.